paletteController.js

```diff
@@ -1,62 +1,28 @@
-class View {
-  constructor(body) {
-    this.body = body;
-  }
-  onKeyPress(cb) {
-    this.body.onkeypress = cb;
-  }
-  onLoad(cb) {
-    this.body.onload = cb;
-  }
-}
-
-class PaletteView {
-  constructor(input, form, container, lockIcon, copyIcon, arrowIcon) {
-    this.input = input;
-    this.form = form;
-    this.container = container;
-    this.lockIcon = lockIcon;
-    this.copyIcon = copyIcon;
-    this.arrowIcon = arrowIcon;
-    this.setupFormAction();
-  }
-
-  setupFormAction() {
-    this.form.onsubmit = (ev) => {
-      ev.preventDefault();
-      this.container.style.backgroundColor = this.input.value;
-    };
-  }
-}
-
 class PaletteController {
   constructor(paletteObjectArr, keyPress) {
     this.paletteObjectArr = paletteObjectArr;
     this.keyPress = keyPress;
-    this.keyPress.onKeyPress(this.paletteShuffle.bind(this));
+    this.keyPress.onKeyPress(this.submit.bind(this));
     this.keyPress.onLoad(this.load.bind(this));
   }
+  paletteShuffle() {
+    for (let i in this.paletteObjectArr) {
+      const arrItem = this.paletteObjectArr[i];
+      if (arrItem.lockState === true) {
+        arrItem.setColor(new Color(randomHex()));
```

Comment:
Why can I call "setColor" in a method from another class? Is it because I'm instantiating a new Color? I thought it would have to be "x = new Color" then "x.setColor." I understood it enough to write it, but after a week, I want to make sure I didn't hack it together.